packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+
+  return (
+    <div className={`${classes.gradientBigTextContainer} ${isScrolledToTop ? 'scrolled-to-top' : ''} ${isScrolledToBottom ? 'scrolled-to-bottom' : ''}`} ref={readPostsContainerRef}>
+      {readPosts.length > 0 ? (
+        readPosts.slice(0, 8).map((post, index) => {
+            return (
+              <PostsTooltip key={index} postId={post._id}>
+                <a key={index} href={"https://www.lesswrong.com/posts/gDijQHHaZzeGrv2Jc/"+post._id}>• {post.title}</a>
+                <br/>
+              </PostsTooltip>
+            )
+          })
+      ) : (
+        <p>(no posts read...)</p>
+      )}
+    </div>
+  );
+};
+
+const Headers = ({ titles, className }: { titles: string[], className: string }) => {
+  return (
+    <>
+      {titles.map((title, index) => (
+        <h5 key={index} className={className}>{title}</h5>
+      ))}
+    </>
+  );
+};
+
+const DialogueCheckBox: React.FC<{
+  targetUserId : string;
+  targetUserDisplayName : string;
+  checkId?: string;
+  isChecked: boolean, 
+  isMatched: boolean;
+  classes: ClassesType;
+}> = ({ targetUserId, targetUserDisplayName, checkId, isChecked, isMatched, classes}) => {
+  const currentUser = useCurrentUser();
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+
+  const [localChecked, setLocalChecked] = useState(false);
+  const [upsertDialogueCheck] = useMutation(gql`
+    mutation upsertUserDialogueCheck($targetUserId: String!, $checked: Boolean!) {
+      upsertUserDialogueCheck(targetUserId: $targetUserId, checked: $checked) {
+          _id
+          __typename
+          userId
+          targetUserId
+          checked
+          match
+        }
+      }
+    `)
+
+  async function handleNewMatchAnonymisedAnalytics() {
+    captureEvent("newDialogueReciprocityMatch", {}) // we only capture match metadata and don't pass anything else
+
+    // ping the slack webhook to inform team of match. YOLO:ing and putting this on the client. Seems fine: but it's the second time this happens, and if we're doing it a third time, I'll properly move it all to the server 
+    const webhookURL = "https://hooks.slack.com/triggers/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";
+    const data = { // Not sending any data for now
+    };
+
+    try {
+      const response = await fetch(webhookURL, {
+        method: 'POST',
+        body: JSON.stringify(data),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+    } catch (error) {
+      //eslint-disable-next-line no-console
+      console.error('There was a problem with the fetch operation: ', error);
+    }
+    
+  }
+
+  const [showConfetti, setShowConfetti] = useState(false);
+
+
+  async function updateDatabase(event: React.ChangeEvent<HTMLInputElement>, targetUserId:string, handleConfetti:React.Dispatch<React.SetStateAction<boolean>>, checkId?:string, ) {
+    if (!currentUser) return;
```

Comment:
Probably just ensure that `currentUser` is not null before doing all of this so that the page is only accessible to logged-in users in the first place and you don't need to do a bunch of redundant null handling.

Example (from the BookmarksPage):
```
  const currentUser = useCurrentUser()
  if (!currentUser) {
    return <span>You must sign in to view this page.</span>
  }
```

Reply:
we hvae that already... but it doesn't get rid of the type error in this location...

Reply:
You have it after this, the type checker isn't smart enough to do time travel (yet)